### PR TITLE
Add Documentation for R4 Procedure Create

### PR DESCRIPTION
**Pull request by nishanthnamana**
The R4 documentation for Procedure Create.

Local Validation:
Procedure_1
Procedure_2

Validation:
Make sure the Procedure page loads and all of the links work properly.

### CODE REVIEW

File: lib/resources/r4/procedure.yaml [hunk 1]
```diff
@@ -49,3 +52,204 @@ fields:
...
+    {
+      "subject": {
+        "reference": "Patient/631923",
+        "display": "Nilsson, Christine"
+      }
+    }
+  url: https://hl7.org/fhir/procedure-definitions.html#Procedure.subject
+  note: subject must be a reference to a Patient.
+
+- name: status
```
Comment (ebaranowski) [inline lib/resources/r4/procedure.yaml:83]: If you list your actions like so then you dont need to repeat in this file: https://github.com/cerner/fhir.cerner.com/pull/371/commits/15b404d166e91e3d9a557e0ad922bfbea002f6c6#diff-beb306fda480ca66704cb041f03846fbR169

Also you should add actions of create

File: lib/resources/r4/procedure.yaml [hunk 2]
```diff
@@ -49,3 +52,204 @@ fields:
...
+  type: Reference
+  description: Indicates the patient who the procedure record is associated with.
+  example: |
+    {
+      "subject": {
+        "reference": "Patient/631923",
+        "display": "Nilsson, Christine"
+      }
+    }
+  url: https://hl7.org/fhir/procedure-definitions.html#Procedure.subject
```
Comment (ebaranowski) [inline lib/resources/r4/procedure.yaml:80]: missing R4

File: lib/resources/r4/procedure.yaml [hunk 3]
```diff
@@ -49,3 +52,204 @@ fields:
...
+  note: subject must be a reference to a Patient.
```
Comment (ebaranowski) [inline lib/resources/r4/procedure.yaml:81]: Just do: Must only include references to Patient.

File: lib/resources/r4/procedure.yaml [hunk 4]
```diff
@@ -49,3 +52,204 @@ fields:
...
+  required: 'Yes'
+  cardinality: 1..1
+  type: code
+  description: Codes identifying the lifecycle stage of an event.
+  example: |
+    {
+      "status": "completed"
+    }
+  url: https://hl7.org/fhir/r4/procedure-definitions.html#Procedure.status
+  note: Only status code <code>completed</code> is supported.
```
Comment (ebaranowski) [inline lib/resources/r4/procedure.yaml:93]: instead of doing code, you can just do ``

File: lib/resources/r4/procedure.yaml [hunk 5]
```diff
@@ -49,3 +52,204 @@ fields:
...
+        {
+          "system": "http://snomed.info/sct",
+          "code": "119954001",
+          "display": "Adenoid excision (procedure)",
+          "userSelected": true
+        }
+      ],
+      "text": "Adenoid excision (text)"
+    }
+  url: https://hl7.org/fhir/procedure-definitions.html#Procedure.code
```
Comment (ebaranowski) [inline lib/resources/r4/procedure.yaml:112]: missing R4

File: lib/resources/r4/procedure.yaml [hunk 6]
```diff
@@ -49,3 +52,204 @@ fields:
...
+  note: A code must be provided.
```
Comment (ebaranowski) [inline lib/resources/r4/procedure.yaml:113]: Must be provided.

File: lib/resources/r4/procedure.yaml [hunk 7]
```diff
@@ -49,3 +52,204 @@ fields:
...
+  cardinality: 0..1
+  type: Reference
+  description: Indicates the encounter when the procedure was performed.
+  example: |
+    {
+      "encounter": {
+        "reference":"Encounter/26070829"
+      }
+    }
+  url: https://hl7.org/fhir/procedure-definitions.html#Procedure.encounter
```
Comment (ebaranowski) [inline lib/resources/r4/procedure.yaml:126]: missing R4

Comment (ebaranowski) [reply lib/resources/r4/procedure.yaml:126]: and below

File: lib/resources/r4/procedure.yaml [hunk 8]
```diff
@@ -49,3 +52,204 @@ fields:
...
+    description: The practitioner who was involved in the procedure.
+    example: |
+      {
+        "actor": {
+          "reference": "Practitioner/2150097",
+          "display": "Strecker, Bradley"
+        }
+      }
+    url: http://hl7.org/fhir/procedure-definitions.html#Procedure.performer.actor
+    note: performer.actor must be a reference to a Practitioner.
```
Comment (ebaranowski) [inline lib/resources/r4/procedure.yaml:236]: Must only include references to Practitioner.

File: lib/resources/r4/procedure.yaml [hunk 9]
```diff
@@ -49,3 +52,204 @@ fields:
...
+  required: 'Yes'
+  cardinality: 1..1
+  type: string
+  description: The type of the FHIR resource.
+  example: |
+    {
+      "resourceType": "Procedure"
+    }
+  url: https://hl7.org/fhir/procedure-definitions.html#Procedure
+  note: resourceType must be Procedure.
```
Comment (ebaranowski) [inline lib/resources/r4/procedure.yaml:66]: also can we change all of the notes to be lists?

File: lib/resources/example_json/r4_examples_procedure.rb [hunk 10]
```diff
@@ -194,5 +194,60 @@ module Resources
         }
       ]
     }
+
+    R4_PROCEDURE_CREATE ||= {
+      "resourceType": "Procedure",
+      "status": "completed",
+      "subject": {
+        "reference": "Patient/1316024",
+        "display": "Tim, Peterson"
```
[the remark this answers is not in the document]
Comment (nishanthnamana) [reply lib/resources/example_json/r4_examples_procedure.rb]: Updated https://github.com/cerner/fhir.cerner.com/pull/372/commits/edd713404a0347648bfd408e254be8a2d11466ce#diff-9cc8375943437b05da52e102f3785695R203

File: lib/resources/r4/procedure.yaml (hunk 1 above)
Comment (nishanthnamana) [reply lib/resources/r4/procedure.yaml]: Updated [here](https://github.com/cerner/fhir.cerner.com/pull/372/commits/edd713404a0347648bfd408e254be8a2d11466ce#diff-80d4a0c17acadba0cd26a39c2cfeaf8dL83)

File: lib/resources/r4/procedure.yaml (hunk 2 above)
Comment (nishanthnamana) [reply lib/resources/r4/procedure.yaml]: Updated [here](https://github.com/cerner/fhir.cerner.com/pull/372/commits/edd713404a0347648bfd408e254be8a2d11466ce#diff-80d4a0c17acadba0cd26a39c2cfeaf8dR94).

File: lib/resources/r4/procedure.yaml (hunk 3 above)
Comment (nishanthnamana) [reply lib/resources/r4/procedure.yaml]: Updated [here](https://github.com/cerner/fhir.cerner.com/pull/372/commits/edd713404a0347648bfd408e254be8a2d11466ce#diff-80d4a0c17acadba0cd26a39c2cfeaf8dR97).

File: lib/resources/r4/procedure.yaml (hunk 4 above)
Comment (nishanthnamana) [reply lib/resources/r4/procedure.yaml]: Updated [here](https://github.com/cerner/fhir.cerner.com/pull/372/commits/edd713404a0347648bfd408e254be8a2d11466ce#diff-80d4a0c17acadba0cd26a39c2cfeaf8dR26).

File: lib/resources/r4/procedure.yaml (hunk 5 above)
Comment (nishanthnamana) [reply lib/resources/r4/procedure.yaml]: Updated [here](https://github.com/cerner/fhir.cerner.com/pull/372/commits/edd713404a0347648bfd408e254be8a2d11466ce#diff-80d4a0c17acadba0cd26a39c2cfeaf8dR34).

File: lib/resources/r4/procedure.yaml (hunk 6 above)
Comment (nishanthnamana) [reply lib/resources/r4/procedure.yaml]: Updated [here](https://github.com/cerner/fhir.cerner.com/pull/372/commits/edd713404a0347648bfd408e254be8a2d11466ce#diff-80d4a0c17acadba0cd26a39c2cfeaf8dR64).

File: lib/resources/r4/procedure.yaml (hunk 7 above)
Comment (nishanthnamana) [reply lib/resources/r4/procedure.yaml]: Updated [here](https://github.com/cerner/fhir.cerner.com/pull/372/commits/edd713404a0347648bfd408e254be8a2d11466ce#diff-80d4a0c17acadba0cd26a39c2cfeaf8dR111).

File: lib/resources/r4/procedure.yaml (hunk 8 above)
Comment (nishanthnamana) [reply lib/resources/r4/procedure.yaml]: Updated [here](https://github.com/cerner/fhir.cerner.com/pull/372/commits/edd713404a0347648bfd408e254be8a2d11466ce#diff-80d4a0c17acadba0cd26a39c2cfeaf8dR243).

File: lib/resources/r4/procedure.yaml (hunk 9 above)
Comment (nishanthnamana) [reply lib/resources/r4/procedure.yaml]: Updated in this [commit](https://github.com/cerner/fhir.cerner.com/pull/372/commits/edd713404a0347648bfd408e254be8a2d11466ce#diff-80d4a0c17acadba0cd26a39c2cfeaf8dR24).

File: lib/resources/example_json/r4_examples_procedure.rb (hunk 10 above)
[the remark this answers is not in the document]
Comment (ebaranowski) [reply lib/resources/example_json/r4_examples_procedure.rb]: But Tim Peters works at cerner... cant you just put in a name like Michael Jordan or Larry Bird?

[the remark this answers is not in the document]
Comment (nishanthnamana) [reply lib/resources/example_json/r4_examples_procedure.rb]: So for reads and searches I saw its 'PETERS, TIMOTHY' for the same id, so update it to that.
Changed it totally [here](https://github.com/cerner/fhir.cerner.com/pull/372/commits/fccbb1a0138b0e19ecee24d550a202f82a4650e5#diff-9cc8375943437b05da52e102f3785695R203).

**Comment by nishanthnamana** [comment]
> Must only include references to Encounter.

My Bad, missed it. Updated [here](https://github.com/cerner/fhir.cerner.com/pull/372/commits/fccbb1a0138b0e19ecee24d550a202f82a4650e5#diff-80d4a0c17acadba0cd26a39c2cfeaf8dR114).

**Comment by nishanthnamana** [comment]
> Must only include references to Location.

Updated [here](https://github.com/cerner/fhir.cerner.com/pull/372/commits/fccbb1a0138b0e19ecee24d550a202f82a4650e5#diff-80d4a0c17acadba0cd26a39c2cfeaf8dR132).

**Comment by ebaranowski** [comment]
Screen Shot 2020-04-07 at 1 25 55 PM
Screen Shot 2020-04-07 at 1 26 51 PM

Validation

File: lib/resources/r4/procedure.yaml [hunk 11]
```diff
@@ -33,19 +47,217 @@ fields:
...
+        }
+      ],
+      "text": "Adenoid excision (text)"
+    }
+  note: |
+    <ul>
+      <li> Must be provided. </li>
+    </ul>
+
+- name: resourceType
```
Comment (ebaranowski) [inline lib/resources/r4/procedure.yaml:67]: Can you order these better? I would prefer the way they are listed on hl7. AKA resourceType should be first.

Comment (nishanthnamana) [reply lib/resources/r4/procedure.yaml]: Updated [here](https://github.com/cerner/fhir.cerner.com/pull/372/commits/a9443bbbf959cd765f514708b4c1c11aeec9b263).

**Comment by MaxPhilips** [comment]
Screen Shot 2020-04-07 at 5 29 14 PM
validated deployment

**Pull request merged**